### DISCUSSION TITLE
Godot python and opencv (and numpy and others)

### DISCUSSION THREAD

**Discussion by TheBricktop** [touilleMan/godot-python#369] [Q&A]
Hello,
im starting a discussion about various ways one can use opencv and numpy (and other image / ai libs) to control and generate content realtime in godot.

First thing probably would be setting up pip and installing opencv :
After some poking around ive managed to use pip via console in VScode,
i've set the interpreter directly to the one provided by godot-python module by copying and pasting the path.
then ive copied and pasted the path one more time to the terminal to access the provided python interpreter and ensure pip via 

path/to/python.exe -m ensurepip

and then installed the opencv and other packages by

path/to/python.exe -m pip install packagename

now ive managed to start the video capture and the webcam has lit so thats somethin'.
But the next step is kinda not so clear for me: how to convert the output of opencv to godot image?
is it possible to do it in a performant way (like realtime ?) 
im referrig this [thread](https://github.com/touilleMan/godot-python/issues/305#issuecomment-913249461) 

its kinda needed for me to make some arvr expirements.

**Comment by Zireael07** [comment]
IIRC opencv output is just normal image files (from when I experimented a bit with it) so should be doable but conversion to imagetexture every time will definitely hurt your performance

**Reply by TheBricktop** [reply]
Maybe there is a way to copy it in gpu or sth cause out of opencv context im only using it for visuals in godot, so no further computations that might involve cpu is needed

so just 

image = Image(frame) 

would be enough ?

@touilleMan  suggested using the PoolByteArray().raw_access() to reuse the numpy array in godot without copying.
